src/sdk/trace/span_processor.rs

```diff
@@ -66,3 +129,215 @@ impl api::SpanProcessor for SimpleSpanProcessor {
+                // Span has finished, add to buffer of pending spans.
+                Some(BatchMessage::ExportSpan(span)) => {
+                    if self.buffer.len() < self.config.max_queue_size {
+                        self.buffer.push(span);
+                    }
+                }
+                // Span batch interval time reached, export current spans.
+                Some(BatchMessage::Tick) => {
+                    if !self.buffer.is_empty() {
+                        let mut spans = std::mem::replace(&mut self.buffer, Vec::new());
+                        while !spans.is_empty() {
+                            let batch_idx = spans
+                                .len()
+                                .saturating_sub(self.config.max_export_batch_size);
+                            let batch = spans.split_off(batch_idx);
+                            self.exporter.export(batch);
+                        }
+                    }
+                }
+                // Stream has terminated or processor is shutdown, return to finish execution.
+                None | Some(BatchMessage::Shutdown) => {
+                    self.exporter.shutdown();
+                    return Poll::Ready(());
+                }
+            }
+        }
+    }
+}
+
+#[derive(Debug)]
+enum BatchMessage {
+    ExportSpan(Arc<exporter::trace::SpanData>),
+    Tick,
+    Shutdown,
+}
+
+impl BatchSpanProcessor {
+    pub(crate) fn new<S, SO, I, IS, ISI>(
+        exporter: Box<dyn exporter::trace::SpanExporter>,
+        spawn: S,
+        interval: I,
+        config: BatchConfig,
+    ) -> Self
+    where
+        S: Fn(BatchSpanProcessorWorker) -> SO,
+        I: Fn(time::Duration) -> IS,
+        IS: Stream<Item = ISI> + Send + 'static,
+    {
+        let (message_sender, message_receiver) = mpsc::channel(config.max_queue_size);
+        let ticker = interval(config.scheduled_delay).map(|_| BatchMessage::Tick);
+
+        // Spawn worker process via user-defined spawn function.
+        spawn(BatchSpanProcessorWorker {
+            exporter,
+            messages: Box::pin(futures::stream::select(message_receiver, ticker)),
+            config,
+            buffer: Vec::new(),
+        });
+
+        // Return batch processor with link to worker
+        BatchSpanProcessor {
+            message_sender: Mutex::new(message_sender),
+        }
+    }
+
+    /// Create a new batch processor builder
+    pub fn builder<E, S, SO, I, IO>(
+        exporter: E,
+        spawn: S,
+        interval: I,
+    ) -> BatchSpanProcessorBuilder<E, S, I>
+    where
+        E: exporter::trace::SpanExporter,
+        S: Fn(BatchSpanProcessorWorker) -> SO,
+        I: Fn(time::Duration) -> IO,
+    {
+        BatchSpanProcessorBuilder {
+            exporter,
+            spawn,
+            interval,
+            config: Default::default(),
+        }
+    }
+}
+
+/// Batch span processor configuration
+#[derive(Debug)]
+pub struct BatchConfig {
```

Comment:
Other implementations also have a _block_ property that won't drop spans and prevents the app from closing until all the spans have been exported.

This can be done with a subsequent PR if preferred.

Reply:
Yeah this could be useful, will add an issue to track 👍